src/status_im/contexts/wallet/add_account/add_address/confirm_address/view.cljs

```diff
@@ -0,0 +1,110 @@
+(ns status-im.contexts.wallet.add-account.add-address.confirm-address.view
+  (:require
+    [clojure.string :as string]
+    [quo.core :as quo]
+    [quo.foundations.colors :as colors]
+    [quo.theme]
+    [react-native.core :as rn]
+    [reagent.core :as reagent]
+    [status-im.common.emoji-picker.utils :as emoji-picker.utils]
+    [status-im.contexts.wallet.add-account.add-address.confirm-address.style :as style]
+    [status-im.contexts.wallet.common.screen-base.create-or-edit-account.view :as
+     create-or-edit-account]
+    [utils.debounce :as debounce]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(def ^:const sheet-closing-delay 750)
+
+(def ^:private confirming-addresses-purposes
+  {:watch {:button-label :t/add-watched-address
+           :address-type :t/watched-address
+           :placeholder  :t/default-watched-address-placeholder}
+   :save  {:button-label :t/save-address
+           :address-type :t/address
+           :placeholder  :t/saved-address}})
+
+(defn view
+  []
+  (let [{:keys [address purpose ens?]} (rf/sub [:get-screen-params])
```

Comment:
'Purpose' here really lacks clarity of the use. It's very generic, is there something more informative we can use?

Reply:
To me it also suggests that we would be better to abstract some pieces of this page but not the whole page, or at least have it configurable from outside rather than encapsulating these properties.
Wdyt @briansztamfater, it was a topic we talked about yesterday